### PR TITLE
Prepare profile first release

### PR DESCRIPTION
**Pull request by quitterie-lcs**
## Purpose

A first release of the LMS profile has to be proposed to the xAPI community.

## Proposal

Documentations have to be cleaned for the release.
The profile is enriched with specified files uploaded/downloaded statements
templates: video, audio, document.

### CODE REVIEW

File: statements/uploaded_document.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+   "object": {
+      "id": "http://gaiax.org/xapi/activities/ba297687-b1aa-4477-9efd-a782c8fdb90a",
+      "definition": {
+         "type": "http://id.tincanapi.com/activitytype/document",
+         "name": {
+            "en": "xAPI 101 Introduction notes"
+         }
+      }
+   },
+   "context": {
```
Comment (quitterie-lcs) [inline statements/uploaded_document.md:29]: Add number of page/sheet ? For example, to get to know if people tend to download if the document is long

File: statements/registered_course.md [hunk 2]
```diff
@@ -26,22 +26,38 @@ The actor is registered to a course.
...
-| Rules | PRESENCE | Remarks |
-|---|---|---|
-| timestamp | INCLUDED | - |
-| context.extensions.http://schema.dases.eu/xapi/profile/common/extension/starting-date | RECOMMENDED | Can be specified when this date is known |
-| context.extensions.http://schema.dases.eu/xapi/profile/common/extension/ending-date | RECOMMENDED | Can be specified when this date is known |
-| context.extensions.http://schema.dases.eu/xapi/profile/common/extension/role | RECOMMENDED | - |
+- `context.contextActivities.category`: INCLUDED, MUST contain an activity with the `https://w3id.org/xapi/lms` id.
+- `context.extensions.https://w3id.org/xapi/lms/extensions/starting-date`: RECOMMENDED, ISO 8601 datetime, can be specified when the date is known.
+- `context.extensions.https://w3id.org/xapi/lms/extensions/ending-date`: RECOMMENDED, ISO 8601 datetime, can be specified when the date is known.
+- `context.extensions.https://w3id.org/xapi/lms/extensions/role`: RECOMMENDED,
```
[the remark this answers is not in the document]
Comment (wilbrdt) [reply statements/registered_course.md:62]: We could describe what the `extensions`.`role` is?

File: statements/uploaded_document.md (hunk 1 above)
Comment (wilbrdt) [reply statements/uploaded_document.md:29]: Do we generally have this information? My guess would be that it is rare to have it

**Review by wilbrdt — COMMENTED** [review]
Alt text

File: statements/downloaded_video.md [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+| Property  | Value         |
+|---|---|
+| verb.id | Must be `http://id.tincanapi.com/verb/downloaded` |
+| object.definition.type | Must be `https://w3id.org/xapi/video/activity-type/video` |
+
+## Rules
+
+- `context.contextActivities.category`: INCLUDED, MUST contain an activity with the `https://w3id.org/xapi/lms` id.
+- `context.extensions.https://w3id.org/xapi/video/extensions/length`: INCLUDED, positive float, MUST contain the value of the time length of the video file.
+- `context.extensions.https://w3id.org/xapi/video/extensions/quality`: INCLUDED, positive integer, MUST contain the value of the quality of the video file.
```
[the remark this answers is not in the document]
Comment (quitterie-lcs) [reply statements/downloaded_video.md:62]: No, you're right. When the video is on a file storage and no player is available, it is not a info we can/would have

File: statements/downloaded_audio.md [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+   "object": {
+      "id": "http://gaiax.org/xapi/activities/ba297687-b1aa-4477-9efd-a782c8fdb90a",
+      "definition": {
+         "type": "https://w3id.org/xapi/video/activity-type/audio",
+         "name": {
+            "en": "xAPI 101 Podcast"
+         }
+      }
+   },
+   "context": {
```
[the remark this answers is not in the document]
Comment (quitterie-lcs) [reply statements/downloaded_audio.md:30]: added as optional fields

**Review by wilbrdt — APPROVED** [review]
LGTM :tada:

**Review by SergioSim — COMMENTED** [review]
Looks Great)
Have only minor questions/suggestions)
image

File: statements/accessed_file.md [hunk 5]
```diff
@@ -30,6 +28,32 @@ This event is emitted when a user has accessed a file such as a pdf, doc, txt, p
...
+
+| Property | Value |
+|---|---|
+| `verb.id` | Must be `https://w3id.org/xapi/netc/verbs/accessed` |
+| `object.definition.type` | Must be `http://activitystrea.ms/file` |
+
+## Rules
+
+- `object.definition.extensions.https://w3id.org/xapi/acrossx/extensions/type`: RECOMMENDED, MUST specify the type of the uploaded file.
+- `context.contextActivities.category`: INCLUDED, MUST contain an activity with the `https://w3id.org/xapi/lms` id.
```
[the remark this answers is not in the document]
Comment (quitterie-lcs) [reply statements/accessed_file.md:58]: I am not sure of that. When looking at the way to build templates on the ADL server, it is not possible to add the profile IRI for this field in the determining properties. 
Furthermore, I would say that it is a property that inherits from the definition of the template itself, as it is defined in this profile. What we have done for virtual classroom profile was to add the field in the statements rules. 
WDYT?

File: statements/accessed_file.md [hunk 6]
```diff
@@ -30,6 +28,32 @@ This event is emitted when a user has accessed a file such as a pdf, doc, txt, p
...
+- `timestamp`: INCLUDED.
```
[the remark this answers is not in the document]
Comment (quitterie-lcs) [reply statements/accessed_file.md:59]: It has been chosen to add the `timestamp` for the profile the Gaia-X working group designed for [this reason](https://github.com/gaia-x-dases/xapi-virtual-classroom/discussions/8)

File: statements/accessed_file.md [hunk 7]
```diff
@@ -30,6 +28,32 @@ This event is emitted when a user has accessed a file such as a pdf, doc, txt, p
...
+## Determining properties
+
+| Property | Value |
+|---|---|
+| `verb.id` | Must be `https://w3id.org/xapi/netc/verbs/accessed` |
+| `object.definition.type` | Must be `http://activitystrea.ms/file` |
+
+## Rules
+
+- `object.definition.extensions.https://w3id.org/xapi/acrossx/extensions/type`: RECOMMENDED, MUST specify the type of the accessed file.
```
[the remark this answers is not in the document]
Comment (quitterie-lcs) [reply statements/accessed_file.md:57]: I think that is a good idea

File: statements/accessed_file.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (SergioSim) [reply statements/accessed_file.md]: Thanks) It's more clear now)

File: statements/downloaded_audio.md [hunk 8]
```diff
@@ -0,0 +1,63 @@
...
+| Property | Value |
+|---|---|
+| `$.verb.id` | MUST be `http://id.tincanapi.com/verb/downloaded` |
+| `$.object.definition.type` | MUST be `https://w3id.org/xapi/video/activity-type/audio` |
+
+
+## Rules
+
+- `$.context.contextActivities.category`: INCLUDED, MUST contain an activity with the `https://w3id.org/xapi/lms` id.
+- `$.context.extensions['https://w3id.org/xapi/video/extensions/length']`: RECOMMENDED, positive float, MUST contain the value of the time length of the audio file.
```
Comment (SergioSim) [inline statements/downloaded_audio.md:61]: Is the length in minutes/seconds/miliseconds? (same question for video length extension)

File: statements/downloaded_video.md [hunk 9]
```diff
@@ -0,0 +1,64 @@
...
+| Property | Value |
+|---|---|
+| verb.id | MUST be `http://id.tincanapi.com/verb/downloaded` |
+| object.definition.type | MUST be `https://w3id.org/xapi/video/activity-type/video` |
+
+## Rules
+
+- `context.contextActivities.category`: INCLUDED, MUST contain an activity with the `https://w3id.org/xapi/lms` id.
+- `context.extensions.https://w3id.org/xapi/video/extensions/length`: RECOMMENDED, positive float, MUST contain the value of the time length of the video file.
+- `context.extensions.https://w3id.org/xapi/video/extensions/quality`: RECOMMENDED, positive integer, MUST contain the value of the quality of the video file.
```
Comment (SergioSim) [inline statements/downloaded_video.md:62]: What metric should be used for the quality? Is it a scale from 0 to 10? The resolution? 
 
In the [video profile](https://liveaspankaj.gitbooks.io/xapi-video-profile/content/statement_data_model.html#2626-quality), the value is a string. e.g. 360, 480, 720, 1080, 2160, sd, hd, 4k, 8k etc.

**Review by SergioSim — APPROVED** [review]
LGTM) :+1: 

image

File: statements/downloaded_audio.md (hunk 8 above)
Comment (quitterie-lcs) [reply statements/downloaded_audio.md:61]: It's in seconds

File: statements/downloaded_video.md (hunk 9 above)
Comment (quitterie-lcs) [reply statements/downloaded_video.md:62]: Their specification have diverged :laughing: 
https://profiles.adlnet.gov/profile/90b2c849-d744-4d0c-8bd0-403e7859a35b/concepts/01ee0d69-3886-47c8-8828-6a55a99ccc81

Comment (quitterie-lcs) [reply statements/downloaded_video.md]: Let's stick to the official profile specification, no? 
https://github.com/adlnet/xapi-authored-profiles/blob/master/video/v1.0.3/video.jsonld#L144-L153

Comment (SergioSim) [reply statements/downloaded_video.md]: Aye, another discrepancy between the video profile recipe and profile on adlnet :disappointed:
However, the [commit](https://github.com/adlnet/xapi-authored-profiles/commit/477340d38e62bfd09630462cfc55f7185ec7c2cd) that added the jsonld profile intended to align with the recipe according to the commit message.
Looking at the [recipe history](https://github.com/liveaspankaj/xapi-video-cop/commit/9c673bbf7c84ea7a571c2e1bb7943a57f85f6311), they indeed changed the quality field type from integer to string in 2017.
Also noticed that we haven't included the `quality` context extension in [ralph's video initialized statement](https://github.com/openfun/ralph/blob/master/src/ralph/models/xapi/video/contexts.py#L95-L130), nor for the video interacted statements - it's only used in the downloaded statement where it's an integer.
In [marsha](https://github.com/openfun/marsha/blob/08afc74a36e1d4677e1adea955e6345e9c9e2881/src/frontend/packages/lib_video/src/components/common/Player/videojs/xapiPlugin/index.ts#L196-L202) the quality field seems to accept both - a string and integer values, although most statements seem to contain an integer value for the quality field.

**Pull request merged**